From ec6682e383def3c24e7545772dbd95e03bafb130 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Tue, 10 Jul 2018 16:00:11 +0800
Subject: [PATCH 161/205] drm: add the struct drm_device platformdev field
 back for sgx drm

This patch is to revert 76adb460 partly since sgx drm use this field.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/tilcdc/tilcdc_drv.c |    1 +
 include/drm/drmP.h                  |    1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/tilcdc/tilcdc_drv.c b/drivers/gpu/drm/tilcdc/tilcdc_drv.c
index d7ae5be..480411c 100644
--- a/drivers/gpu/drm/tilcdc/tilcdc_drv.c
+++ b/drivers/gpu/drm/tilcdc/tilcdc_drv.c
@@ -245,6 +245,7 @@ static int tilcdc_init(struct drm_driver *ddrv, struct device *dev)
 	if (IS_ERR(ddev))
 		return PTR_ERR(ddev);
 
+	ddev->platformdev = pdev;
 	ddev->dev_private = priv;
 	platform_set_drvdata(pdev, ddev);
 	drm_mode_config_init(ddev);
diff --git a/include/drm/drmP.h b/include/drm/drmP.h
index fcdee70..16f26d7 100644
--- a/include/drm/drmP.h
+++ b/include/drm/drmP.h
@@ -439,6 +439,7 @@ struct drm_device {
 	struct pci_controller *hose;
 #endif
 
+	struct platform_device *platformdev; /**< Platform device struture */
 	struct virtio_device *virtdev;
 
 	struct drm_sg_mem *sg;	/**< Scatter gather memory */
-- 
1.7.9.5

